#161589741 Implement landing page design

olusoladavid opened the pull request:
#### What does this PR do?
- Implements landing page design and functionality.

#### Description of Task to be completed?

- [x] - Add unit and integration tests
- [x] - Add responsive header
- [x] - Add responsive ArticleListView with article cards
- [x] - Add responsive TagListView
- [x] - Fetch and render articles from backend into ArticleListView
- [x] - Fetch and render tags from backend into TagListView


#### How should this be manually tested?
Run the following commands sequentially:
- `git fetch`
- `git checkout ft-161589741-landing-page`
- `npm install`
- Rename the `.env.sample` file on the root directory of the cloned repo to `.env`. Uncomment and supply the following variable: `REACT_APP_HOST` which is the link to the frontend application; `REACT_APP_BASE_API_URL` which is the base URL for the backend api. For example, `https://odin-ah-backend-staging.herokuapp.com/api/v1`.
- `npm start`
- The landing page should be displayed by default

#### What are the relevant pivotal tracker stories?
[#161589741](www.pivotaltracker.com/n/projects/2197379/stories/161589741)

#### Screenshots (if appropriate)
<img width="1438" alt="screen shot 2018-11-15 at 12 01 03 pm" src="https://user-images.githubusercontent.com/30433120/48548803-2ff08900-e8ce-11e8-886f-6694e679a6e6.png">

waleCloud reviewed (commented):
Increase test coverage, endeavor to cover all necessary files created/modified. before raising a final PR.
well done so far! 👍

olusoladavid commented:
Thank you for the feedback @waleCloud . I will work on improving the test coverage by covering every component, action and reducer I created or modified.

olusoladavid committed:
feat(landing-page): add responsive header and articles section

- add unit and integration tests
- refactor end-to-end tests
- add responsive header
- add responsive ArticleListView with article cards
- add responsive TagListView
- fetch and render articles from backend into ArticleListView
- fetch and render tags from backend into TagListView

[Delivers #161589741]

uchemukolo reviewed (commented):
@olusoladavid Great job with implementing the landing page. The PR was tested locally and the page is responsive on all devices. while I was testing, I noticed that is the user is redirected to a blank page after logging in. please fix it
